### PR TITLE
[FT-4738] Merge `develop` into `procurify-development`

### PR DESCRIPTION
**Pull request by mtucciarone**
[FT-4738](https://procurifyteam.atlassian.net/browse/FT-4738)

6 criticals were discovered:
CVE-2022-34265 - django
CVE-2020-7471 - django
CVE-2022-28347 - django
CVE-2023-31047 - django
CVE-2022-28346 - django
GHSA-4ppp-gpcr-7qf6 - waitress

Work required: 

Remediate the security vulnerabilities in [GitHub - procurify/django-star-ratings: Star ratings for your Django models with a single template tag. Python 3 compatible.](https://github.com/procurify/django-star-ratings) and in the branch develop

Merge `develop` into `procurify-development` branch

Update the `procurifydevelopment` repo file `pyproject.tml` and update the commit rev to the commit of this PR merge commit

[FT-4738]: https://procurifyteam.atlassian.net/browse/FT-4738?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mtucciarone** [comment]
This requires https://github.com/procurify/django-star-ratings/pull/21 to be merged in as well

**Comment by steve-curran** [comment]
@mtucciarone  Please add a description to provide context for these changes and a ticket. 

This repo settings is also missing branch protection, which can result in unreviewed commits, which I believe is a violation of our SoC controls.

**Review by steve-curran — CHANGES_REQUESTED** [review]
Missing description - branch protection rules need to be updated. See comment [above.](https://github.com/procurify/django-star-ratings/pull/23#issuecomment-1932801487)

**Comment by mtucciarone** [comment]
@steve-curran i don't have rights on this repo to that degree -- @neilpower can probably help here.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+# https://docs.github.com/github/administering-a-repository/configuration-options-for-dependency-updates
+
+version: 2
+updates:
+  - package-ecosystem: "pip" # See documentation for possible values
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+      time: "08:00"
+  - package-ecosystem: "npm" # See documentation for possible values
```
Comment (steve-curran) [inline .github/dependabot.yml:13]: Why do we need npm here? 

I see a package.json that refers to a sass build process but I don't see any of the associated directories. Is there a pipeline that runs for this repo to produce assets?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+      time: "08:00"
+  - package-ecosystem: "bundler" # See documentation for possible values
```
Comment (steve-curran) [inline .github/dependabot.yml:18]: Does this repo have ruby deps?

File: star_ratings/views.py [hunk 3]
```diff
@@ -25,21 +27,22 @@ def get_object(self):
     def post(self, request, *args, **kwargs):
         def _post(request, *args, **kwargs):
             data = request.POST or json.loads(request.body.decode())
-
+            
+            data = copy.deepcopy(data)
             return_url = data.pop('next', '/')
+            if type(return_url) == list:
+                return_url = return_url[0]
```
Comment (steve-curran) [inline star_ratings/views.py:34]: What happens when the list is empty?

File: tests/base.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
+from django.apps import apps
+from django.conf import settings
+
+
+class BaseFooTest:
+    """
+        We need to use a different Foo class if the model rating model is switched out (as it needs UUID).
```
Comment (steve-curran) [inline tests/base.py:7]: What is a `Foo` class?

File: tests/base.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    """
+    def setUp(self):
+        foo_app, foo_model_name = settings.FOO_MODEL.split('.')
+        self.foo_model = apps.get_model(app_label=foo_app, model_name=foo_model_name)
+
+        bar_app, bar_model_name = settings.BAR_MODEL.split('.')
+        self.bar_model = apps.get_model(app_label=bar_app, model_name=bar_model_name)
```
Comment (steve-curran) [inline tests/base.py:14]: missing newline terminator here as flagged by github. Same on many other files

**Review by steve-curran — COMMENTED** [review]
Some comments. 

I've very little context to review this PR other than it aims to resolve vulnerabilities in dependencies. 

e.g - I see behavior changes [here](https://github.com/procurify/django-star-ratings/pull/23/files#diff-baa8e65b68e5642f3466cf656dcffedee24abdd414e96d78d88e1a6e5aded686R44)  but I do not know why they are being made.

**Comment by nav** [comment]
Hey Procurify team, I am apparently subscribed to this repo. @steve-curran For context, this was a fork of https://pypi.org/project/django-star-ratings/ done by Vinny (IIRC) so most of the code is from the parent repo.

**Comment by neilpower** [comment]
@steve-curran @mtucciarone - We have branch protection rules for the default branch `develop` but not for the branch `procurify-development`. Should we be merging into `develop` here instead or should our default branch be updated?

**Comment by steve-curran** [comment]
> Hey Procurify team, I am apparently subscribed to this repo. @steve-curran For context, this was a fork of https://pypi.org/project/django-star-ratings/ done by Vinny (IIRC) so most of the code is from the parent repo.

Thanks @nav, I appreciate the context here. - I'll work with @neilpower to figure out your lingering access.

**Comment by mtucciarone** [comment]
@neilpower it would make sense to just use `develop` as the branch that `procurifydevelopment` refers to in its packages for pulling in `django-star-ratings`

**Comment by mtucciarone** [comment]
@steve-curran Since `develop` has been the maintained branch for this repo, there's been some upstreams pulls from the fork to ensure that our version upgrades and bugfixes from the fork also make their way into this codebase and repository. About 8 months ago I did an update to this code to remediate all of the github security issues which came with code changes from the fork merge. I did not divert from any of the incoming code nor did I explicitly change any code functionality in said pull. Since this repo is being targeted for deletion in our sunsetting plan, I'd like to focus on the scope here of removing our vulnerabilities over code quality that we're not maintaining at the moment. 

While I understand the need to perform an uplift to this code and to comply to our quality standards, we currently have no plans to make this a priority over security vulnerability remediation.

**Comment by steve-curran** [comment]
> @neilpower it would make sense to just use `develop` as the branch that `procurifydevelopment` refers to in its packages for pulling in `django-star-ratings`

@mtucciarone - this would make sense to me

**Review by steve-curran — APPROVED** [review]
After getting more context in backend chapter meeting I'm ok to approve this. 

I would like to see us get off this as soon as possible. I've added an item to our 1:1 @neilpower to see where we are at in this regard.

**Pull request merged**